src/dune_rules/workspace.ml

```diff
@@ -462,6 +462,14 @@ module Context = struct
       Common.equal base t.base && Opam_switch.equal switch t.switch
     ;;
 
+    let name_hint_opt name =
+      if name <> "" && String.contains name '/'
```

Comment:
The test for `""` is unnecessary, but what do you think about just checking at the beginning for absolute paths? (`String.is_prefix ~prefix:"/"`)

Reply:
Is it necessary an absolute or could it work with relative path?

Reply:
I think that opam switch names are either absolute paths (local switches) or strings without `/` (global switches), but there might be other code paths that try to build context names from user input.

Reply:
I think we can keep the heuristic that it starts with a `/` then as the last scenario seems uncommon.